spec/models/gestionnaire_spec.rb

```diff
@@ -78,4 +79,105 @@
+      let!(:follow_commentaire_groupe_gestionnaire_unread_commentaire_cause_seen_at_before_last_commentaire) { create(:follow_commentaire_groupe_gestionnaire, groupe_gestionnaire: groupe_gestionnaire, gestionnaire: gestionnaire_unread_commentaire_cause_seen_at_before_last_commentaire, sender: administrateur, commentaire_seen_at: 1.day.ago) }
+
+      it do
+        expect(gestionnaire.unread_commentaires?(groupe_gestionnaire)).to eq false
+        expect(gestionnaire_unread_commentaire_cause_never_seen.unread_commentaires?(groupe_gestionnaire)).to eq true
+        expect(gestionnaire_unread_commentaire_cause_seen_at_before_last_commentaire.unread_commentaires?(groupe_gestionnaire)).to eq true
+      end
+    end
+  end
+
+  describe "#commentaire_seen_at" do
+    context "when already seen commentaire" do
+      let(:gestionnaire) { create(:gestionnaire) }
+      let(:administrateur) { create(:administrateur) }
+      let(:groupe_gestionnaire) { create(:groupe_gestionnaire, gestionnaires: [gestionnaire]) }
+      let!(:commentaire_groupe_gestionnaire) { create(:commentaire_groupe_gestionnaire, groupe_gestionnaire: groupe_gestionnaire, sender: administrateur, created_at: 12.hours.ago) }
+      let!(:follow_commentaire_groupe_gestionnaire) { create(:follow_commentaire_groupe_gestionnaire, groupe_gestionnaire: groupe_gestionnaire, gestionnaire: gestionnaire, sender: administrateur, commentaire_seen_at: Time.zone.now) }
+
+      it { expect(gestionnaire.commentaire_seen_at(groupe_gestionnaire, administrateur)).not_to eq nil }
+    end
+
+    context "when never seen commentaire" do
+      let(:gestionnaire) { create(:gestionnaire) }
+      let(:administrateur) { create(:administrateur) }
+      let(:groupe_gestionnaire) { create(:groupe_gestionnaire, gestionnaires: [gestionnaire]) }
+      let!(:commentaire_groupe_gestionnaire) { create(:commentaire_groupe_gestionnaire, groupe_gestionnaire: groupe_gestionnaire, sender: administrateur, created_at: 12.hours.ago) }
+
+      it { expect(gestionnaire.commentaire_seen_at(groupe_gestionnaire, administrateur)).to eq nil }
+    end
+  end
+
+  describe "#mark_commentaire_as_seen" do
+    context "when already seen commentaire" do
+      let(:now) { Time.zone.now.beginning_of_minute }
+      let(:gestionnaire) { create(:gestionnaire) }
+      let(:administrateur) { create(:administrateur) }
+      let(:groupe_gestionnaire) { create(:groupe_gestionnaire, gestionnaires: [gestionnaire]) }
+      let!(:commentaire_groupe_gestionnaire) { create(:commentaire_groupe_gestionnaire, groupe_gestionnaire: groupe_gestionnaire, sender: administrateur, created_at: 12.hours.ago) }
+      let!(:follow_commentaire_groupe_gestionnaire) { create(:follow_commentaire_groupe_gestionnaire, groupe_gestionnaire: groupe_gestionnaire, gestionnaire: gestionnaire, sender: administrateur, commentaire_seen_at: 12.hours.ago) }
+
+      before do
```

Comment:
```suggestion
      subject do
```
il faudra appeler subject dans le `it`

La motivation c'est que d'une part, le subject désigne le code qu'on veut tester, et d'autre part on se rend compte d'un point de vue maintenance des specs, il vaut mieux séparer before & subject lorsque les tests se complexifient au fil du temps (là encore, notre codebase est loin d'être clean!)

Reply:
done